frame/assets/src/lib.rs

```diff
@@ -237,12 +238,13 @@ pub mod pallet {
 	#[pallet::storage]
 	/// Approved balance transfers. First balance is the amount approved for transfer. Second
 	/// is the amount of `T::Currency` reserved for storing this.
-	pub(super) type Approvals<T: Config<I>, I: 'static = ()> = StorageDoubleMap<
+	pub(super) type Approvals<T: Config<I>, I: 'static = ()> = StorageNMap<
 		_,
-		Blake2_128Concat,
-		T::AssetId,
-		Blake2_128Concat,
-		ApprovalKey<T::AccountId>,
+		(
+			storage::Key<Blake2_128Concat, T::AssetId>,
+			storage::Key<Blake2_128Concat, T::AccountId>,
+			storage::Key<Blake2_128Concat, T::AccountId>,
+		),
```

Comment:
maybe we can extract this change in a follow-up PR, so then we can tag `runtime-migration` on a minimal PR just changing this pallet.

Reply:
yeah, this will pretty soon be used in statemint and I think we should extract this part into another PR.